File: seed/ruby-sdk-v2/trace/lib/seed/types/v2test_case_with_actual_result_implementation.rb
# frozen_string_literal: true

module Seed
  module Types
    class V2TestCaseWithActualResultImplementation < Internal::Types::Model
      field :get_actual_result, -> { Seed::Types::V2NonVoidFunctionDefinition }, optional: false, nullable: false, api_name: "getActualResult"
      field :assert_correctness_check, -> { Seed::Types::V2AssertCorrectnessCheck }, optional: false, nullable: false, api_name: "assertCorrectnessCheck"
    end
  end
end

File: seed/php-sdk/trace/src/Traits/V2TestCaseWithActualResultImplementation.php
<?php

namespace Seed\Traits;

use Seed\Types\V2NonVoidFunctionDefinition;
use Seed\Types\V2AssertCorrectnessCheckZero;
use Seed\Types\V2AssertCorrectnessCheckOne;
use Seed\Core\Json\JsonProperty;
use Seed\Core\Types\Union;

/**
 * @property V2NonVoidFunctionDefinition $getActualResult
 * @property (
 *    V2AssertCorrectnessCheckZero
 *   |V2AssertCorrectnessCheckOne
 * ) $assertCorrectnessCheck
 */
trait V2TestCaseWithActualResultImplementation
{
    /**
     * @var V2NonVoidFunctionDefinition $getActualResult
     */
    #[JsonProperty('getActualResult')]
    public V2NonVoidFunctionDefinition $getActualResult;

    /**
     * @var (
     *    V2AssertCorrectnessCheckZero
     *   |V2AssertCorrectnessCheckOne
     * ) $assertCorrectnessCheck
     */
    #[JsonProperty('assertCorrectnessCheck'), Union(V2AssertCorrectnessCheckZero::class, V2AssertCorrectnessCheckOne::class)]
    public V2AssertCorrectnessCheckZero|V2AssertCorrectnessCheckOne $assertCorrectnessCheck;
}
